src/azul/service/index_query_service.py

```diff
@@ -80,13 +87,19 @@ def get_data(self,
         if item_id is not None:
             validate_uuid(item_id)
             filters['entryId'] = {'is': [item_id]}
-        if source_ids is not None:
-            self._add_implicit_sources_filter(filters, source_ids)
+        if source_filter is not None and not source_filter.indicate_only:
+            self._add_implicit_sources_filter(filters, source_filter.source_ids)
         response = self.transform_request(catalog=catalog,
                                           filters=filters,
                                           pagination=pagination,
                                           entity_type=entity_type)
 
+        if source_filter is not None and source_filter.indicate_only:
+            for hit in response['hits']:
+                entity = one(hit[entity_type])
+                entity_source_id = one(hit['sources'])['sourceId']
```

Comment:
```suggestion
                source_id = one(hit['sources'])['sourceId']
```

unless there is a conflicting variable. LMK if there is.